### DISCUSSION TITLE
What is the best way to access HTTP API？

### DISCUSSION THREAD

**Discussion by znoliver** [eclipse-theia/theia#13474] [Q&A] [answered]
I want to access HTTP API in my theia browser extension ,what is the best way? Use Ajax/Fetch in extension？Use RPC to node backend and node access http?Do you have any good examples？

**Comment by msujew** [comment, marked as answer]
Hey @znoliver,

Theia doesn't impose anything on adopters regarding HTTP traffic. However, I can recommend you to use the injectable `RequestService`. It will respect system HTTP proxy settings when perform any HTTP call. The service can be injected in both the frontend and the backend of the app, although injecting it on the frontend is usually better.

**Reply by znoliver** [reply]
thank you very much
